src/Tribe/Aggregator/Processes/Service_Provider.php

```diff
@@ -25,6 +30,8 @@ public function register() {
 
 		$this->handle_clear_request();
 		$this->handle_clear_result();
+
+		tribe( Tribe__Events__Aggregator__Processes__Batch_Imports::class );
```

Comment:
Since this is registered as a singleton above and then resolved in the context of this method, I would refactor this to  a faster form:

```php
$batch_imports = new Tribe__Events__Aggregator__Processes__Batch_Imports();
$batch_imports->hook();
tribe_singleton(Tribe__Events__Aggregator__Processes__Batch_Imports::class, $batch_imports);
```

This would avoid the resolution logic in the container that would pay off only in the instance of a non eager resolution, as is the case here.

Reply:
I've suggested a better approach for this above, though. Where the class is not managing any hooking at all.